lib/libutee/tee_api_operations.c

```diff
@@ -810,17 +826,30 @@ TEE_Result TEE_DigestDoFinal(TEE_OperationHandle operation, const void *chunk,
 
 /* Cryptographic Operations API - Symmetric Cipher Functions */
 
-void TEE_CipherInit(TEE_OperationHandle operation, const void *IV, uint32_t IVLen)
+void TEE_CipherInit(TEE_OperationHandle operation, void *IV, uint32_t IVLen)
 {
 	TEE_Result res;
 
 	if (operation == TEE_HANDLE_NULL)
 		TEE_Panic(0);
+
 	if (operation->info.operationClass != TEE_OPERATION_CIPHER)
 		TEE_Panic(0);
+
+	if (!(operation->info.handleState & TEE_HANDLE_FLAG_KEY_SET) ||
+	    !(operation->key1))
+		TEE_Panic(0);
+
+	if (operation->operationState != TEE_OPERATION_STATE_INITIAL)
+		TEE_ResetOperation(operation);
+
+	if (operation->operationState == TEE_OPERATION_STATE_INITIAL)
```

Comment:
This `if ()` is not needed because `TEE_ResetOperation()` always sets `operationState` to `INITIAL` (or panics).